min_heap_binary_tree.rb

```diff
@@ -15,6 +15,27 @@ def insert(node)
     set_min_heap_state
   end
 
+  def delete(node)
+    replacement_node = tree[-1]
+    delete_index = tree.find_index(node)
+    swap! delete_index, -1
+    tree.pop
+
+    if parent(delete_index).value > replacement_node.value
+      swap! delete_index, parents_index(delete_index)
+    elsif left_child(delete_index) != nil && right_child(delete_index) != nil
```

Comment:
It looks like the case for having **only a left child**, still needs to be addressed.  Both in the specs and the code.